src/client/render2d/pixi_renderer.tsx

```diff
@@ -0,0 +1,119 @@
+import * as classNames from 'classnames'
+import { IReactionDisposer } from 'mobx'
+import { observable } from 'mobx'
+import { action } from 'mobx'
+import { autorun } from 'mobx'
+import { observer } from 'mobx-react'
+import { WebGLRenderer } from 'pixi.js'
+import { CanvasRenderer } from 'pixi.js'
+import { Container } from 'pixi.js'
+import { autoDetectRenderer } from 'pixi.js'
+import * as React from 'react'
+import { Component } from 'react'
+import ReactResizeDetector from 'react-resize-detector'
+
+import { Transform } from '../math/transform'
+
+import { pixiObject } from './pixi/rendering'
+import { RendererProps } from './renderer_props'
+import * as style from './style.css'
+
+@observer
+export class PixiRenderer extends Component<RendererProps> {
+  @observable private resolution: Transform = Transform.of()
+  private canvas: HTMLCanvasElement
+  private renderer: WebGLRenderer | CanvasRenderer
+  private stopAutorun: IReactionDisposer
+
+  componentDidMount() {
+    if (!this.canvas) {
+      return
+    }
+
+    this.renderer = autoDetectRenderer({
+      view: this.canvas,
+      transparent: true,
+      antialias: true,
+    })
+
+    // Render when changes happen
+    this.stopAutorun = autorun(this.renderCanvas, {
+      scheduler: requestAnimationFrame,
+    })
+  }
+
+  componentWillUnmount() {
+    if (this.stopAutorun) {
+      this.stopAutorun()
+    }
+  }
+
+  render() {
+    return (
+      <div className={classNames(this.props.className, style.container)}>
+        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} />
+        <canvas
+          className={style.container}
+          width={-this.resolution.translate.x * 2}
+          height={-this.resolution.translate.y * 2}
+          ref={this.onRef}
+        />
+      </div>
+    )
+  }
+
+  private onRef = (canvas: HTMLCanvasElement) => {
```

Comment:
This should be `canvas: HTMLCanvasElement | null`, I'm not sure how its passing the build without this.